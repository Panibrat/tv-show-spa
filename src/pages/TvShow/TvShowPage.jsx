import React, { Component } from 'react';
import styles from './styles.module.css';
import EpisodesList from '../../components/EpisodesList';

export class TvShowPage extends Component {
    componentDidMount() {
        const { fetchTvShows, fetchEpisodesList } = this.props;
        fetchTvShows();
        fetchEpisodesList();
    }

    render() {
        const { data, isFetching, episodesList } = this.props;
        if (isFetching) {
            return (
                <h1>Fetching...</h1>
            )
        }

        return (
            <div className={ styles.container }>
                <div className={ styles.title}>
                    <h1>{ data.name }</h1>
                </div>
                <div className={ styles.content }>
                    <div className={ styles.coverImage }>
                        <img src={ data && data.image && data.image.original ? data.image.original : '#' }
                             alt={ data.name }
                        />
                    </div>
                    <div className={ styles.info }>
                        <div className={ styles.description }>
                            <h3>Description</h3>
                            <span
                                dangerouslySetInnerHTML={{
                                __html: data.summary
                            }}>
                            </span>
                        </div>
                        <div className={ styles.episodesContainer }>
                            <h3>Episodes</h3>
                            <div className={ styles.episodesList }>
                                <EpisodesList episodesList = { episodesList } />
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
