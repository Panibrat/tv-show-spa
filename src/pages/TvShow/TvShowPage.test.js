import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TvShowPage } from './TvShowPage';

jest.mock('../../components/EpisodesList', () => {
    return () => 'Episode list';
});

describe('TvShowPage Component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            data: {
                name: null,
                image: {
                    original: '#'
                },
                summary: null,
            },
            episodesList: [],
            isFetching: false,
            fetchTvShows: jest.fn(),
            fetchEpisodesList: jest.fn(),
        };
        wrapper = shallow(<TvShowPage {...props} />);
    });

    it('should renders correctly with empty data', () => {
        const wrapper = mount(<TvShowPage {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should renders correctly fetching state', () => {
        wrapper.setProps(
            {
                isFetching: true,
            }
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should renders correctly with filled data', () => {
        wrapper.setProps(
            {
                data: {
                    name: 'Test name',
                    image: {
                        original: 'url'
                    },
                    summary: 'Test Summary',
                },
                episodesList: [
                    {
                        id: 0,
                        name: 'Test E name 0'
                    },
                    {
                        id: 1,
                        name: 'Test E name 1'
                    },
                ],
                isFetching: false,
                fetchTvShows: jest.fn(),
                fetchEpisodesList: jest.fn(),
            }
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
