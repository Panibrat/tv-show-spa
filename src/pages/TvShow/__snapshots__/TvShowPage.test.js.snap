// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowPage Component should renders correctly fetching state 1`] = `
<h1>
  Fetching...
</h1>
`;

exports[`TvShowPage Component should renders correctly with empty data 1`] = `
<TvShowPage
  data={
    Object {
      "image": Object {
        "original": "#",
      },
      "name": null,
      "summary": null,
    }
  }
  episodesList={Array []}
  fetchEpisodesList={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  fetchTvShows={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isFetching={false}
>
  <div
    className="container"
  >
    <div
      className="title"
    >
      <h1 />
    </div>
    <div
      className="content"
    >
      <div
        className="coverImage"
      >
        <img
          alt={null}
          src="#"
        />
      </div>
      <div
        className="info"
      >
        <div
          className="description"
        >
          <h3>
            Description
          </h3>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": null,
              }
            }
          />
        </div>
        <div
          className="episodesContainer"
        >
          <h3>
            Episodes
          </h3>
          <div
            className="episodesList"
          >
            <Component
              episodesList={Array []}
            >
              Episode list
            </Component>
          </div>
        </div>
      </div>
    </div>
  </div>
</TvShowPage>
`;

exports[`TvShowPage Component should renders correctly with filled data 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    <h1>
      Test name
    </h1>
  </div>
  <div
    className="content"
  >
    <div
      className="coverImage"
    >
      <img
        alt="Test name"
        src="url"
      />
    </div>
    <div
      className="info"
    >
      <div
        className="description"
      >
        <h3>
          Description
        </h3>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Test Summary",
            }
          }
        />
      </div>
      <div
        className="episodesContainer"
      >
        <h3>
          Episodes
        </h3>
        <div
          className="episodesList"
        >
          <Component
            episodesList={
              Array [
                Object {
                  "id": 0,
                  "name": "Test E name 0",
                },
                Object {
                  "id": 1,
                  "name": "Test E name 1",
                },
              ]
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
